libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -763,6 +763,51 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (ASTPointer<Expression const> const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += ", " + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+		m_context.subObjects().insert(contract);
+
+		Whiskers t(R"(
+			let memPos := <allocate>(0)
```

Comment:
You cannot use fixed variable names outside of helper functions.

Reply:
Actually `allocate(0)` is a good idea :)

Maybe we should make that more explicit by introducing `allocate_unbounded()`?

Maybe `unbounded` is not the right word, it's more like unbounded and not reserved.

Reply:
Maybe even adding size as argument to protect of overflow
I did like that, hope it is fine

Reply:
The point is that we usually don't know the size at that point in time. So it looks like we need two new functions :)